doc/nrf/ug_contribution.rst

```diff
@@ -0,0 +1,127 @@
+Contribution Workflow
+*********************
+
+One general practice we encourage, is to make small,
+controlled changes. This practice simplifies review, makes merging and
+rebasing easier, and keeps the change history clear and clean.
+
+When contributing to NCS, it is also important you provide as much
+information as you can about your change, update appropriate documentation,
+and test your changes thoroughly before submitting.
+
+The general GitHub workflow used by NCS developers uses a combination of
+command line Git commands and browser interaction with GitHub.  As it is with
+Git, there are multiple ways of getting a task done.  We'll describe a typical
+workflow here:
+
+.. _Create a Fork of NCS manifest repo (fw-nrfconnect-nrf):
+  https://github.com/NordicPlayground/fw-nrfconnect-nrf#fork-destination-box
+
+#. `Create a Fork of fw-nrfconnect-nrf`_
+   to your personal account on GitHub. (Click on the fork button in the top
+   right corner of the fw-nrfconnect-nrf repo page in GitHub.)
+
+#. On your development computer, change into the :file:`nrf` folder that was
+   created when you :ref:`obtained the code <get_the_code>`::
+
+     cd ncs/nrf
+
+   Rename the default remote pointing to the `upstream repository
+   <https://github.com/NordicPlayground/fw-nrfconnect-nrf>`_ from ``origin`` to
+   ``upstream``::
+
+     git remote rename origin upstream
+
+   Let Git know about the fork you just created, naming it ``origin``::
```

Comment:
Do this before renaming the remote above, makes sense to create it before "using" it

Reply:
I am a bit unsure on how you want this setup in the doc.